internal/relationtuple/transact_server.go

```diff
@@ -52,6 +52,23 @@ func (h *handler) TransactRelationTuples(ctx context.Context, req *acl.TransactR
 	}, nil
 }
 
+func (h *handler) DeleteRelationTuples(ctx context.Context, req *acl.DeleteRelationTuplesRequest) (*acl.DeleteRelationTuplesResponse, error) {
+	if req.Query == nil {
+		return nil, errors.New("invalid request")
+	}
+
+	q, err := (&RelationQuery{}).FromProto((*acl.ListRelationTuplesRequest_Query)(req.Query))
```

Comment:
The generated proto definitions have getter functions, so you can instead rewrite `func (q *RelationQuery) FromProto(...)` to take an interface with the getter functions. That way you don't have to convert the type which might break in the future at some point.